src/components/workflow/workflow.service.ts

```diff
@@ -0,0 +1,866 @@
+            }),
+            relation('out', '', 'nextPossibleState', {
+              active: true
+            }),
+            node('nextState')
+          ]
+        ])
+        .return({
+          nextState:[
+            { id: 'id' },
+            { value: 'value' }
+          ]
+        })
+        .run()) as State[];
+
+      return { items: result.filter(item => item.id && item.value) };
+
+    } catch (e) {
+      this.logger.warning('Failed to delete state', {
+        exception: e,
+      });
+
+      throw e;
+    }
+  };
+
+  // attachSecurityGroupToState
+  async attachSecurityGroup(session: ISession, input: GroupState): Promise<void>{
+    try{
+      await this.db
+        .query()
+        .match([
+          [
+            node('token', 'Token', {
+              active: true,
+              value: session.token
+            }),
+            relation('in', '', 'token', {
+              active: true,
+            }),
+            node('requestingUser', 'User', {
+              id: session.userId
+            }),
+            relation('in', '', 'member'),
+            node('sg', 'SecurityGroup', {
+              id: input.securityGroupId
+            }),
+          ],
+          [
+            node('state', 'State', {
+              id: input.stateId
+            })
+          ]
+        ])
+        .merge([
+          [
+            node('state'),
+            relation('out', '', 'securityGroup', {
+              active: true
+            }),
+            node('sg')
+          ]
+        ])
+        .first();
+
+    } catch (e) {
+      this.logger.warning('could not attach security group to state', {
+        exception: e
+      });
+      throw e;
+    }
+  };
+
+  // removeSecurityGroupFromState
+  async removeSecurityGroup(session: ISession, input: GroupState): Promise<void>{
+    try{
+      await this.db
+        .query()
+        .match([
+          [
+            node('token', 'Token', {
+              active: true,
+              value: session.token
+            }),
+            relation('in', '', 'token', {
+              active: true,
+            }),
+            node('requestingUser', 'User', {
+              id: session.userId
+            }),
+            relation('in', '', 'member'),
+            node('sg', 'SecurityGroup', {
+              id: input.securityGroupId
+            })
+          ],
+          [
+            node('state', 'State', {
+              id: input.stateId
+            }),
+            relation('out', 'rel', 'securityGroup'),
+            node('sg')
+          ]
+        ])
+        .detachDelete('rel')
+        .first();
+
+    } catch (e) {
+      this.logger.warning('could not remove security group from state', {
+        exception: e
+      });
+      throw e;
+    }
+  }
+
+  // we are using security groups as notification groups for now
+  // attachNotificationGroupToState
+  async attachNotificationGroup(session: ISession, input: GroupState): Promise<void>{
```

Comment:
for this kind of addition, let's just require the requesting user to be an admin of the base node. They don't need to be an admin of the security group.